lib/js/asyncjs.nim

```diff
@@ -83,20 +83,46 @@ proc replaceReturn(node: var NimNode) =
       replaceReturn(son)
     inc z
 
+proc isFutureVoid(node: NimNode): bool =
+  result = node.kind == nnkBracketExpr and
```

Comment:
I am aware this might be not entirely perfect, as a user might alias `Future[void]`? 

On the other hand that doesn't work with asyncdispatch too, so I guess it should be ok

Reply:
You should be reusing as much of the code in asyncdispatch as possible: https://github.com/nim-lang/Nim/blob/devel/lib/pure/asyncmacro.nim#L313

Which is why this code should have been in asyncdispatch in the first place...